libpay-braintree/src/main/scala/com/wix/pay/braintree/JsonBraintreeOAuthMerchantParser.scala

```diff
@@ -0,0 +1,16 @@
+package com.wix.pay.braintree
+
+import org.json4s.DefaultFormats
+import org.json4s.native.Serialization
+
+class JsonBraintreeOAuthMerchantParser() extends BraintreeMerchantParser {
```

Comment:
If accept my comment regarding type parameterized parser trait, you can check if this class is required altogether.
I'm not sure about what I'm about to suggest, with erasure and all, but give it a try:
If you change the current implementation of BraintreeMerchantParser to something like:
class JsonBraintreeMerchantParser[T] extends BraintreeMerchantParser[T] {
   override def parse(merchantKey: String): T = {
      Serialization.read[T](merchantKey)
   }

   override def stringify(merchant: T): String = {
      Serialization.write(merchant)
   }
}

Then this class may not be required.